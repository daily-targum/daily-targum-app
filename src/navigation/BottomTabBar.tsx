import React from 'react';
import { View, Platform } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler'
import { useNavigation } from '@react-navigation/core';
import { Surface, Theme, Section } from '../components';
import { navigation as navConfig } from '../constants';
import { BottomTabBarProps } from '@react-navigation/bottom-tabs';

export const HEIGHT = navConfig.bottomTabBar.height;

function Footer(props: BottomTabBarProps) {
  const {insets, colors} = Theme.useTheme();
  const navigation = useNavigation();
  return (
    <Surface
      tint='dark'
      style={[props.style, {
        position: 'absolute',
        bottom: 0,
        left: 0,
        right: 0,
        height: navConfig.bottomTabBar.height + insets.bottom
      }]}
      backgroundTint={colors.primary}
    >
      <Section 
        style={{
          flex: 1,
          marginBottom: insets.bottom,
          borderTopWidth: 0.5,
          borderTopColor: colors.divider
        }}
        innerStyle={{
          padding: 0,
          flexDirection: 'row',
          justifyContent: 'space-around',
          alignItems: 'center'
        }}
      >
        {Object.keys(props.descriptors).map((d, i) => {
          const descriptor: any = props.descriptors[d];
          return (
            <TouchableOpacity
              key={d}
              onPress={() => navigation.navigate({ key: d })}
            >
              {descriptor.options.tabBarIcon({
                size: 24,
                focused: i === props.state.index,
                // TODO: check how react navigation has this configured
                color: i === props.state.index ? props.activeTintColor : props.inactiveTintColor
              })}
            </TouchableOpacity>
          );
        })}
      </Section>
    </Surface>
  );
}

function useHeight({
  safe = true
}: {
  safe?: boolean
}) {
  const {insets} = Theme.useTheme();

  const paddingBottom = (safe ? insets.bottom : 0);
  return navConfig.bottomTabBar.height + paddingBottom;
}

function getUnsafeHeight(): number {
  return navConfig.bottomTabBar.height;
}

function ScrollSpacer() {
  const paddingBottom = useHeight({safe: true});
  return (
    <View style={{
      height: Platform.OS !== 'ios' ? paddingBottom : 0
    }}/>
  );
}

Footer.useHeight = useHeight;
Footer.getUnsafeHeight = getUnsafeHeight;
Footer.ScrollSpacer = ScrollSpacer;
export default Footer;