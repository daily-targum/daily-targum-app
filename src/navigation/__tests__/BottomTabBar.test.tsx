jest.mock('../../components/Theme');

import React from 'react';
import renderer from 'react-test-renderer';
import BottomTabBar, { HEIGHT as HEADER_HEIGHT } from '../BottomTabBar';

describe('BottomTabBar', () => {

  it('useHeight', () => {
    let consumedHeight;
    function ReadHeight() {
      consumedHeight = BottomTabBar.useHeight({safe: false});
      return null;
    }
    renderer.create(
      <ReadHeight/>
    );
    expect(consumedHeight).toBe(HEADER_HEIGHT);
  });

  it('<ScrollSpacer />', () => {
    const tree = renderer.create(
      <BottomTabBar.ScrollSpacer/>
    );
    expect(tree).toMatchSnapshot();
  });

});
