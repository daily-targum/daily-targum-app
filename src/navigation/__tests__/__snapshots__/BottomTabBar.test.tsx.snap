// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomTabBar <ScrollSpacer /> 1`] = `
<View
  style={
    Object {
      "height": 0,
    }
  }
/>
`;
